spectrocloud/resource_cluster_coxedge.go

```diff
@@ -565,29 +565,31 @@ func resourceCoxEdgeClusterUpdate(ctx context.Context, d *schema.ResourceData, m
 
 		for _, mp := range ns {
 			machinePoolResource := mp.(map[string]interface{})
-			name := machinePoolResource["name"].(string)
-			hash := resourceMachinePoolCoxEdgeHash(machinePoolResource)
+			if machinePoolResource["name"].(string) != "" {
```

Comment:
```suggestion
                       // since known issue in TF SDK: https://github.com/hashicorp/terraform-plugin-sdk/issues/588
			if machinePoolResource["name"].(string) != "" {
```